### PR TITLE
DOCU-3662: improve plugin versioning

### PR DESCRIPTION
**Pull request by fabianrbz**
### Description

Related [Jira ticket](https://konghq.atlassian.net/browse/DOCU-3662)

Add support for `minimum_version` and `maximum_version` in
the frontmatter of plugin pages.

* If a page has one of these set in frontmatter, it gets generated
for that version range, similar to versions.yml
* If a page doesn’t have any minimum or maximum version,
it defaults to the setting for the whole plugin, which is set in versions.yml

The only caveat is that we had to replace `_metadata.yml` with a specific
`_metadata` folder. If a file for a specific version exists, (e.g.
`_metadata/_2.6.x.yml`) it will be used for that specific release, if
not `_metadata/_index.yml` will be used.

I also fixed a bug in the releases dropdown.
When switching to an older version for which the current page does not exist, it redirected the user to the plugin's landing page in its latest version instead of the version selected by the user.

https://github.com/Kong/docs.konghq.com/assets/715229/526c0144-2fd2-40df-b433-3769d50805e1

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

### CODE REVIEW

**Comment by fabianrbz** [comment]
@lena-larionova I updated some of the docs, but we should write about the new behavior somewhere. what do you think about a new section [here](https://docs.konghq.com/contributing/single-sourced-plugins/)?

File: app/contributing/single-sourced-plugins.md
Comment (lena-larionova) [inline app/contributing/single-sourced-plugins.md:1]: Should we also add something about putting `minimum_version` or `maximum_version` in doc front matter?

**Review by lena-larionova — CHANGES_REQUESTED** [review]
Some minor content changes.

File: app/contributing/single-sourced-plugins.md
Comment (lena-larionova) [reply app/contributing/single-sourced-plugins.md:1]: I just saw your comment 
>  I updated some of the docs, but we should write about the new behavior somew[here](https://docs.konghq.com/contributing/single-sourced-plugins/). what do you think about a new section here?

Let's add a new section there. I think that page needs an overhaul anyway, but we don't really have another place for it.

**Comment by lena-larionova** [comment]
I had no issues with any of the versioning, it works! It seems far more intuitive this way. And I love to see this:
Screenshot 2024-02-29 at 10 16 16 AM

I think @Guaris should also take a look at this, as he had to version one of the plugins the old way before and had trouble - so I'd love to see his feedback on this approach over the old.

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 nav_title: Overview
 ---
 
+{% if_plugin_version gte:3.1.x %}
```
[the remark this answers is not in the document]
Comment (fabianrbz) [reply app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md:5]: what about the `3.0.x` ones?

**Comment by Guaris** [comment]
LGTM

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lena-larionova) [reply app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md]: Ack - sorry I missed this. Same thing, they don't need to exist.

**Review by lena-larionova — APPROVED** [review]
LGTM

**Pull request merged**